src/devices/Mcp23xxx/Mcp23008.cs

```diff
@@ -48,23 +61,75 @@ public byte[] Read(Register.Address startingRegisterAddress, byte byteCount)
         }
 
         /// <summary>
-        ///  Writes a byte to a register.
+        /// Sets a mode to a pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number.</param>
+        /// <param name="mode">The mode to be set.</param>
+        public void SetPinMode(int pinNumber, PinMode mode)
+        {
+            ValidateMode(mode);
+            ValidatePin(pinNumber);
+            _ioDirection = ChangePin(
+                Register.Address.IODIR,
+                pinNumber,
+                mode == PinMode.Input ? PinValue.High : PinValue.Low,
+                Read(Register.Address.IODIR));
```

Comment:
shouldn't this be _ioDirection instead? If not, then what is the point of caching?

Reply:
Caching was only to add validation and throw in your face if you write to an input. I'm going to put up an issue for implementing the feature.

Reply:
Is there anything other than simply changing this line for implementing that feature? If not, I would rather just do it now. I'm fine on doing it later if there are other concerns like multithreading and such.

Reply:
I would have to add the check to all the places that need it and refresh it when the device is reset- it isn't one line.